java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -1120,6 +1118,23 @@ private boolean hasLocallyPresentOptionalBinding(ResolvedBindings resolvedBindin
               && !getLocalExplicitBindings(keyFactory.unwrapOptional(resolvedBindings.key()).get())
                   .isEmpty();
         }
+
+        /**
+         * Returns {@code true} if {@code resolvedBindings} contains a binding for which there is
+         * an explicit present binding in this component that was resolved previously.
+         */
+        boolean hasLocalContributionBinding(ResolvedBindings resolvedBindings) {
+          return resolvedBindings
+                .contributionBindings()
+                .stream()
+                .anyMatch(contributionBinding ->
+                    !getLocalExplicitBindings(resolvedBindings.key()).isEmpty()
+                  && parentResolver.isPresent()
+                  && parentResolver.get().getPreviouslyResolvedBindings(
+                      resolvedBindings.bindingKey()).isPresent()
+                  && !parentResolver.get().getPreviouslyResolvedBindings(
```

Comment:
ef7c84e0 demonstrates an idea I had to address the owning component problem described above. This should pass Travis.

Let me know what you think!